nengo_gui/components/netgraph.py

```diff
@@ -125,12 +125,15 @@ def _reload(self, code=None):
             if new_uid != uid:
                 new_item = None
 
+            same_class = (isinstance(new_item, old_item.__class__) or
+                          isinstance(old_item, new_item.__class__))
+
             # find reasons to delete the object.  Any deleted object will
             # be recreated, so try to keep this to a minimum
             keep_object = True
             if new_item is None:
                 keep_object = False
-            elif not isinstance(new_item, old_item.__class__):
+            elif same_class:
```

Comment:
Yikes...   really good catch finding this.  That logic was definitely wrong.....  

Do you think something like this might be more robust?

```
for cls in [nengo.Ensemble, nengo.Node, nengo.Network, nengo.Connection]:
    if isinstance(new_item, cls) and isinstance(old_item, cls):
        same_class = True
        break
else:
    same_class = False
```